### PR TITLE
[#181] Test systemd services in VM

### LINKED ISSUE #181: Test generated systemd services
# Clarification and motivation
Since #156 we're going to provide quite complex systemd services and it'd be good to have some automatic tests for them in CI.

# Acceptance criteria
CI tests systemd services provided by the native packages.

### PR DESCRIPTION
**Pull request by rvem**
## Description

This PR introduces tests for systemd services provided by our `.deb` packages with Octez binaries.

Currently, there are tests for `tezos-node`, `tezos-baking`, `tezos-baker`, and `tezos-signer` services.
At the moment, they can only be run locally.
For usage instructions refer to this [README](https://github.com/serokell/tezos-packaging/tree/rvem/%23181-systemd-services-testing/tests/systemd).

Roadmap:
1) ~Integrate these tests into CI. For this, we'll first need to merge #480 to have a fast way of building binary `.deb` packages~
2) ~Provide tests for more services. Perhaps, make these tests a bit safer.~
3) ~Test non-default services configuration~

## Related issue(s)

Resolves #181

#### Related changes (conditional)

- [x] I checked whether I should update the [README](/serokell/tezos-packaging/tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

**Comment by krendelhoff2** [comment]
`default: ===================== 3 failed, 8 passed in 183.20 seconds =====================`
` default: ===================== 4 failed, 7 passed in 225.83 seconds =====================`
`default: ===================== 3 failed, 8 passed in 183.20 seconds =====================`
it keeps failing, always same error
```
    default:   Failed to acquire the protocol version from the node
    default:   Rpc request failed:
    default:      - meth: GET
    default:      - uri: http://localhost:8732/chains/main/blocks/head/protocols
    default:      - error: Unable to connect to the node: "Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")"
```
actually I usually faced it during tezos-setup-wizard debug, it used to resolve after restart

**Comment by krendelhoff2** [comment]
why you did it here? should i change `tezos-node` service in new release?

**Comment by rvem** [comment]
@krendelhoff2, may I ask you to try services with changes from #523 with these tests to check if they're more stable now

**Comment by krendelhoff2** [comment]
worked fine for me now

**Review by krendelhoff2 — COMMENTED** [review]
i forgot to submit the review 2 months ago

File: tests/systemd/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  config.vm.provision "file", source: "#{packagesDirectory}", destination: "$HOME/out"
+  config.vm.provision "file", source: "services_tests.py", destination: "$HOME/services_tests.py"
+  config.vm.provision "shell", inline: <<-SHELL
+    add-apt-repository -y ppa:serokell/tezos && apt-get update
+    apt-get install -y tezos-sapling-params acl python3-pystemd python3-pytest python3-psutil
+  SHELL
+  config.vm.provision "shell", path: "bootstrap.sh"
+  config.vm.provision "shell", inline: <<-SHELL
+    pytest-3 services_tests.py
+  SHELL
```
[the remark this answers is not in the document]
Comment (rvem) [reply tests/systemd/Vagrantfile:22]: Sadly, in case of tests fail with non-zero exit code, all bootstrap will have to be run from scratch, because the entire VM is deleted, but this should indeed optimize the case when tests succeed

**Review by krendelhoff2 — APPROVED** [review]
LGTM

File: tests/systemd/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<!--
+   - SPDX-FileCopyrightText: 2022 Oxhead Alpha
```
Comment (pasqu4le) [inline tests/systemd/README.md:2]: Minor thing, but since these have been modified recently and not yet merge, could you bump the year of all the files?

File: tests/systemd/services_tests.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            f"octez-client import secret key remote-signer {signer_backends[service_type]}{remote_key} --force"
+        )
+        proc_call("octez-client --mode mockup sign bytes 0x1234 for remote-signer")
+
+
+def test_node_mainnet_service():
+    node_service_test("mainnet")
+
+
+def test_node_limanet_service():
```
Comment (pasqu4le) [inline tests/systemd/services_tests.py:120]: limanet has been decommissioned at this point, can you update these to mumbainet instead?

**Review by pasqu4le — COMMENTED** [review]
I have only a couple of comments, but IIUC you still intend to add these tests to CI, right?

**Comment by rvem** [comment]
> but IIUC you still intend to add these tests to CI, right?

Yes, at least I hope to do so

File: tests/systemd/services_tests.py [hunk 4]
```diff
@@ -128,16 +128,16 @@ def test_node_mainnet_service():
     node_service_test("mainnet")
 
 
-def test_node_limanet_service():
-    node_service_test("limanet")
+def test_node_mumbainet_service():
```
Comment (krendelhoff2) [inline tests/systemd/services_tests.py:131]: There changes seem to be unrelated to the commit
Otherwise LGTM

**Review by krendelhoff2 — APPROVED** [review]

File: tests/systemd/services_tests.py (hunk 4 above)
Comment (rvem) [reply tests/systemd/services_tests.py:131]: My rebasing skills are far from ideal :see_no_evil:
